### PR TITLE
TS Linting react-router

### PR DESCRIPTION
**Pull request by DenisCarriere**
- Using the default `dtslint/dt.json` for linting

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dslint/dt.json" }`.

### CODE REVIEW

File: types/react-router/index.d.ts [hunk 1]
```diff
@@ -15,112 +15,82 @@
...
+export interface RouteComponentProps<P> {
+  match: match<P>;
+  location: H.Location;
+  history: H.History;
+}
 
-  function matchPath<P>(pathname: string, props: RouteProps): match<P> | null;
+export interface RouteProps {
+  location?: H.Location;
+  component?: React.SFC<RouteComponentProps<any> | undefined> | React.ComponentClass<RouteComponentProps<any> | undefined>;
```
Comment (erkieh) [inline types/react-router/index.d.ts]: Shouldn't these undefined be any instead?

Comment (erkieh) [inline types/react-router/index.d.ts]: Shouldn't these undefined be any instead?

Comment (DenisCarriere) [inline types/react-router/index.d.ts]: This line wasn't modified by this PR: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/16240/files#diff-bc5f906a3af0cde1a95a8a23cb993e96L68

The `react-router` definition needs some cleaning to do, @erkieh don't hesitate to send a new PR with the proposed changes.

**Comment by tkrotoff** [comment]
Did not see this PR

I wanted to keep the examples as close as possible to the [official ones](https://github.com/ReactTraining/react-router/tree/22bd52c901ef0312348c2b02549c7102bd5653ae/packages/react-router-website/modules/examples) so that they are easy to update/maintain in sync.

That's why there is no semicolons in my original PR #15170

Linting index.d.ts is good, linting everything is stupid.
This is not specific to react-router: other definitions use copy-pasted examples as tests and you want to keep them as close as possible to the original ones.

**Pull request merged**